diff --git a/src/components/Siderba/index.jsx b/src/components/Siderba/index.jsx
index faf71be..8a65cce 100644
--- a/src/components/Siderba/index.jsx
+++ b/src/components/Siderba/index.jsx
@@ -1,7 +1,7 @@
 import { Row, Col, List, Card } from 'antd';
 // import {List } from '@ant-design/icons';
 import './Siderba.css';
-import { getRateListAction ,getListHotelAction} from '../../redux/actions';
+import { getRateListAction} from '../../redux/actions';
 import { connect } from 'react-redux';
 import { useState,useEffect } from 'react';
 import { Rate } from 'antd';
@@ -13,114 +13,59 @@ function Siderba({
   // getHotelList,
   getRateList,
   rateList,
-  // listHotel
+  getListHotelByRate,
+  listHotel
   
 }) {
 
-  const [locationSelected, setLocationSelected] = useState(0);
+  const [rateSelected, setRateSelected] = useState(0);
   useEffect(() => {
-    // getRateList();
-    getRateList({
-      page: 1,
-      limit: 4,
-      
-    });
+    getRateList();
   }, []);
 
 
   function handleFilterLocation(id) {
-    console.log("🚀 ~ file: index.jsx ~ line 20 ~ handleFilterLocation ~ id", id)
-    setLocationSelected(id);
-    getRateList({
-      page: 1,
-      limit: 4,
-      rateId: id,
-
-    });
+    setRateSelected(id);
+    getListHotelByRate(id);
   }
 
   return (
     <>
       <Row gutter={16} style={{ padding: '0 16px' }}>
-        {/* <Col span={24} >
-          <Card
-            hoverable
-            style={{ width: 240 }}
-            cover={<img alt="example" src="https://os.alipayobjects.com/rmsportal/QBnOOoLaAfKPirc.png" />}
-          >
-            <Meta title="Europe Street beat" description="www.instagram.com" />
-          </Card>,
-       </Col> */}
         <Col span={24}>
           <List
             size="small"
             header={<div>Tìm kiếm </div>}
             bordered
-            dataSource={[
-              ...rateList.data,
-            ]}
-            renderItem={(item) => (
+            dataSource={[ 1, 2, 3, 4, 5 ]}
+            renderItem={(item) => 
+              (
               <List.Item className ="list"
-                onClick={() => handleFilterLocation(item.id)}
-                style={{ color: locationSelected === item.id ? 'red' : 'black' }}
+                onClick={() => handleFilterLocation(item)}
+                style={{ color: rateSelected === item ? 'red' : 'black' }}
               >     
-                  <Rate disabled defaultValue={item.rate} /> 
+                  <Rate disabled defaultValue={item} /> 
                  
               </List.Item>
             )}
           />
         </Col>
       </Row>
-      {/* <Row gutter={16} style={{ padding: '0 16px' }}>
-        {/* <Col span={24} >
-          <Card
-            hoverable
-            style={{ width: 240 }}
-            cover={<img alt="example" src="https://os.alipayobjects.com/rmsportal/QBnOOoLaAfKPirc.png" />}
-          >
-            <Meta title="Europe Street beat" description="www.instagram.com" />
-          </Card>,
-       </Col> */}
-        {/* <Col span={24}>
-          <List
-            size="small"
-            header={<div>Tìm kiếm </div>}
-            bordered
-            dataSource={[
-              ...listHotel.data.rooms,
-            ]}
-            renderItem={(item) => (
-              <List.Item
-                onClick={() => handleFilterLocation(item.id)}
-                style={{ color: locationSelected === item.id ? 'red' : 'black' }}
-              >     
-                  {item.name} 
-                 
-              </List.Item>
-            )}
-          />
-        </Col>
-      </Row> */} 
-
     </>
 
   )
 }
 const mapStateToProps = (state) => {
-  const { rateList } = state.productHotelReducer;
-  // const { listHotel } = state.productHotelReducer;
-  console.log("🚀 ~ file: index.jsx ~ line 69 ~ mapStateToProps ~ rateList", rateList)
-  
+  const { rateList, listHotel } = state.productHotelReducer;
   return {
     rateList: rateList,
-    // listHotel: listHotel,
+    listHotel: listHotel,
   }
 };
 
 const mapDispatchToProps = (dispatch) => {
   return {
     getRateList: (params) => dispatch(getRateListAction(params)),
-    // getHotelList: (params) => dispatch(getListHotelAction(params)),
   };
 }
 
diff --git a/src/db/db.json b/src/db/db.json
index f40acf1..3b1aebd 100644
--- a/src/db/db.json
+++ b/src/db/db.json
@@ -301,7 +301,7 @@
     },
     {
       "id": 5,
-      "locationId": 1,
+      "locationId": 2,
       "name": "Hotel Supper Godel",
       "img": "https://pix10.agoda.net/hotelImages/626432/-1/4e377e4b246a20370ac4a9fb2e4850be.jpg?s=1024x768",
       "rate": 5,
@@ -340,7 +340,7 @@
       "combo": "3N2D + Buffet + Đưa đón sân bay+ Massage"
     },
     {
-      "id": 1,
+      "id": 9,
       "locationId": 2,
       "img": "https://cdn.huongnghiepaau.com/wp-content/uploads/2019/10/khach-san-pullman-ha-noi.jpg",
       "rate": 5,
@@ -350,7 +350,7 @@
       "combo": "2N2D + Buffet + Đưa đón sân bay"
     },
     {
-      "id": 2,
+      "id": 10,
       "locationId": 2,
       "img": "https://pix10.agoda.net/hotelImages/4885116/0/3ec7a573097742323e2f976670239467.jpg?s=1024x768",
       "rate": 4,
@@ -360,7 +360,7 @@
       "combo": "3N2D + Buffet + Đưa đón sân bay+ Massage"
     },
     {
-      "id": 3,
+      "id": 11,
       "locationId": 2,
       "img": "https://toquoc.mediacdn.vn/2019/2/28/my-trieu-summit-15513252506751703394469.jpg",
       "rate": 4,
@@ -370,7 +370,7 @@
       "combo": "2N2D + Buffet + Đưa đón sân bay"
     },
     {
-      "id": 4,
+      "id": 12,
       "locationId": 2,
       "name": "Hotel Mường Thanh",
       "img": "https://cf.bstatic.com/images/hotel/max1024x768/863/86327330.jpg",
@@ -380,7 +380,7 @@
       "combo": "3N2D + Buffet + Đưa đón sân bay+ Massage"
     },
     {
-      "id": 5,
+      "id": 13,
       "locationId": 2,
       "img": "https://cdnimg.vietnamplus.vn/uploaded/fsmsr/2015_11_26/movenpickhanoi.jpg",
       "rate": 5,
@@ -390,7 +390,7 @@
       "combo": "2N2D + Buffet + Đưa đón sân bay"
     },
     {
-      "id": 6,
+      "id": 14,
       "locationId": 2,
       "img": "https://bazantravel.com/cdn/medias/uploads/30/30991-khach-san-sheraton-hanoi.jpg",
       "rate": 5,
@@ -400,7 +400,7 @@
       "combo": "3N2D + Buffet + Đưa đón sân bay+ Massage"
     },
     {
-      "id": 7,
+      "id": 15,
       "locationId": 2,
       "img": "https://ik.imagekit.io/tvlk/apr-asset/TzEv3ZUmG4-4Dz22hvmO9NUDzw1DGCIdWl4oPtKumOg=/hotels/1000000/530000/523600/523534/246b00cd_z.jpg?tr=q-40,c-at_max,w-740,h-500&_src=imagekit",
       "rate": 3,
@@ -410,7 +410,7 @@
       "combo": "2N2D + Buffet + Đưa đón sân bay"
     },
     {
-      "id": 8,
+      "id": 16,
       "locationId": 2,
       "img": "https://baogialai.com.vn/dataimages/202009/original/images2927238_logo_img20191128112544_1_15991305286161571887111.jpg",
       "rate": 2,
@@ -420,7 +420,7 @@
       "combo": "3N2D + Buffet + Đưa đón sân bay+ Massage"
     },
     {
-      "id": 1,
+      "id": 17,
       "locationId": 4,
       "img": "https://cdn.huongnghiepaau.com/wp-content/uploads/2019/10/khach-san-pullman-ha-noi.jpg",
       "rate": 5,
@@ -430,7 +430,7 @@
       "combo": "2N2D + Buffet + Đưa đón sân bay"
     },
     {
-      "id": 2,
+      "id": 18,
       "locationId": 4,
       "img": "https://pix10.agoda.net/hotelImages/4885116/0/3ec7a573097742323e2f976670239467.jpg?s=1024x768",
       "rate": 4,
@@ -440,7 +440,7 @@
       "combo": "3N2D + Buffet + Đưa đón sân bay+ Massage"
     },
     {
-      "id": 3,
+      "id": 19,
       "locationId": 4,
       "img": "https://toquoc.mediacdn.vn/2019/2/28/my-trieu-summit-15513252506751703394469.jpg",
       "rate": 4,
@@ -450,7 +450,7 @@
       "combo": "2N2D + Buffet + Đưa đón sân bay"
     },
     {
-      "id": 4,
+      "id": 20,
       "locationId": 4,
       "name": "Hotel Mường Thanh",
       "img": "https://cf.bstatic.com/images/hotel/max1024x768/863/86327330.jpg",
@@ -460,7 +460,7 @@
       "combo": "3N2D + Buffet + Đưa đón sân bay+ Massage"
     },
     {
-      "id": 5,
+      "id": 21,
       "locationId": 4,
       "img": "https://cdnimg.vietnamplus.vn/uploaded/fsmsr/2015_11_26/movenpickhanoi.jpg",
       "rate": 5,
@@ -470,7 +470,7 @@
       "combo": "2N2D + Buffet + Đưa đón sân bay"
     },
     {
-      "id": 6,
+      "id": 22,
       "locationId": 4,
       "img": "https://bazantravel.com/cdn/medias/uploads/30/30991-khach-san-sheraton-hanoi.jpg",
       "rate": 3,
@@ -480,7 +480,7 @@
       "combo": "3N2D + Buffet + Đưa đón sân bay+ Massage"
     },
     {
-      "id": 7,
+      "id": 23,
       "locationId": 4,
       "img": "https://ik.imagekit.io/tvlk/apr-asset/TzEv3ZUmG4-4Dz22hvmO9NUDzw1DGCIdWl4oPtKumOg=/hotels/1000000/530000/523600/523534/246b00cd_z.jpg?tr=q-40,c-at_max,w-740,h-500&_src=imagekit",
       "rate": 3,
@@ -490,14 +490,14 @@
       "combo": "2N2D + Buffet + Đưa đón sân bay"
     },
     {
-      "id": 8,
+      "id": 24,
       "locationId": 4,
-      "img": "https://baogialai.com.vn/dataimages/202009/original/images2927238_logo_img20191128112544_1_15991305286161571887111.jpg",
-      "rate": 2,
-      "name": "Hotel Lõ Hưng",
-      "Title": "Đường Phạm Hưng",
-      "Price": "5.190.000 VND",
-      "combo": "3N2D + Buffet + Đưa đón sân bay+ Massage"
+      "img": "https://ik.imagekit.io/tvlk/apr-asset/TzEv3ZUmG4-4Dz22hvmO9NUDzw1DGCIdWl4oPtKumOg=/hotels/1000000/530000/523600/523534/246b00cd_z.jpg?tr=q-40,c-at_max,w-740,h-500&_src=imagekit",
+      "rate": 3,
+      "name": "Hotel Linh Dương",
+      "Title": "Đường Trần Phú",
+      "Price": "4.690.000 VND",
+      "combo": "2N2D + Buffet + Đưa đón sân bay"
     }
   ],
   "rooms": [
diff --git a/src/pages/ListHotel/index.jsx b/src/pages/ListHotel/index.jsx
index 7575f7e..bc74065 100644
--- a/src/pages/ListHotel/index.jsx
+++ b/src/pages/ListHotel/index.jsx
@@ -1,4 +1,4 @@
-import { Button, Card, Col, Row } from 'antd';
+import { Button, Card, Col, Row, List } from 'antd';
 
 import { connect } from 'react-redux';
 import { getListHotelAction } from '../../redux/actions';
@@ -9,17 +9,14 @@ import './styles.css';
 import { ThunderboltOutlined } from '@ant-design/icons';
 import SearchTourPage from '../SearchTour';
 
-import Siderba from '../../components/Siderba';
-
-
 function ListHotelPage({
   listHotel,
   getListHotel,
   match,
 }) {
+  const [categorySelected, setCategorySelected] = useState(undefined);
   const locationId = match.params.id;
   const [roomSelected, setRoomSelected] = useState({});
- 
   
   useEffect(() => {
    
@@ -36,7 +33,16 @@ function ListHotelPage({
     }
   }, [listHotel.data])
 
-   
+
+  function handleFilterLocation(id) {
+    setCategorySelected(id);
+    getListHotel({
+      page: 1,
+      limit: 4,
+      id: locationId,
+      rate: id
+    });
+  }
 
   function loadmoreHotel(){
     getListHotel({
@@ -44,12 +50,13 @@ function ListHotelPage({
       // page: page + 1,
       page: listHotel.page + 1,
       limit: 4,
-      locationId: roomSelected,
+      id: locationId,
+      rate: categorySelected
     });
   }
 
   function renderListHotel() {
-    return listHotel.data.hotels.map((item, index) => {
+    return listHotel.data.map((item, index) => {
       return (
         <>
           <Row gutter={[12, 12]}>
@@ -81,6 +88,7 @@ function ListHotelPage({
         </>
       )
     });
+    
   }
 
   return (
@@ -88,12 +96,30 @@ function ListHotelPage({
       <SearchTourPage/>
       <Row gutter={[8, 8]} justify="center">
         <Col span={7}>
-          < Siderba />
+          <Row gutter={16} style={{ padding: '0 16px' }}>
+            <Col span={24}>
+              <List
+                size="small"
+                header={<div>Tìm kiếm </div>}
+                bordered
+                dataSource={[ 1, 2, 3, 4, 5 ]}
+                renderItem={(item) => 
+                  (
+                  <List.Item className ="list"
+                    onClick={() => handleFilterLocation(item)}
+                  >     
+                      <Rate disabled defaultValue={item} /> 
+                    
+                  </List.Item>
+                )}
+              />
+            </Col>
+          </Row>
         </Col>
         <Col span={17}>
           {renderListHotel()}
         </Col>
-        {listHotel.data.hotels.length % 4 === 0 && (
+        {listHotel.data.length % 4 === 0  && (
             <Button onClick={()=>loadmoreHotel()}>Xem thêm khách sạn</Button>
         )}
   
@@ -111,32 +137,11 @@ const mapStateToProps = (state) => {
 
 const mapDispatchToProps = (dispatch) => {
   return {
-    getListHotel: (params) => dispatch(getListHotelAction(params)),
+    getListHotel: (params) => dispatch(getListHotelAction(params))
   };
 }
 
 export default connect(mapStateToProps, mapDispatchToProps)(ListHotelPage);
 
 
-  // const productDetail = productList.find((item) => item. hotelsid.toString() ===  hotelsid);
-  // return (
-  //   <>
-  //   <Row >
-  //     <Col span ={16}>
-  //   <Card>
-
-  //     <img style={{width:200 , height:200}} src={productDetail.img} />
-  //     <div className="detail">
-  //     <h2>
-  //     {productDetail.name}
-  //     </h2>
-  //     <span>{productDetail.comment}</span>
-  //     </div>
-  //   </Card>
-  //     </Col>
-  //   </Row>
-  //   </>
-  // );
-
-
 
diff --git a/src/redux/actions/product.action.js b/src/redux/actions/product.action.js
index b3ded6a..b51bc5d 100644
--- a/src/redux/actions/product.action.js
+++ b/src/redux/actions/product.action.js
@@ -30,3 +30,4 @@ export function getProductLocationListAction(params) {
       payload: params,
     }
   }
+  
diff --git a/src/redux/reducers/product.reducer.js b/src/redux/reducers/product.reducer.js
index 887aceb..81ada3d 100644
--- a/src/redux/reducers/product.reducer.js
+++ b/src/redux/reducers/product.reducer.js
@@ -5,9 +5,7 @@ const initialState = {
       error: '',
     },
     listHotel: {
-      data: {
-        hotels:[],
-      },
+      data: [],
       load: false,
       error: '',
     },
@@ -67,7 +65,7 @@ const initialState = {
           },
         }
       }
-
+      
       case 'GET_LIST_HOTEL_REQUEST': {
         return {
           ...state,
@@ -78,16 +76,46 @@ const initialState = {
         }
       }
       case 'GET_LIST_HOTEL_SUCCESS': {
+        
+        // const { data } = action.payload;
+        // return {
+        //   ...state,
+        //   listHotel: {
+        //     ...state.listHotel,
+        //     data: data,
+        //     load: false,
+        //   },
+        // }
 
-        const { data } = action.payload;
-        return {
-          ...state,
-          listHotel: {
-            ...state.listHotel,
-            data: data,
-            load: false,
-          },
+
+
+        const { data, page, more } = action.payload;
+        console.log("🚀 ~ file: product.reducer.js ~ line 93 ~ productHotelReducer ~ data", data)
+        if (more) {
+          return {
+            ...state,
+            listHotel: {
+              ...state.listHotel,
+              data: [
+                ...state.listHotel.data,
+                ...data,
+              ],
+              page: page,
+              load: false,
+            },
+          }
+        } else {
+          return {
+            ...state,
+            listHotel: {
+              ...state.listHotel,
+              data: data,
+              page: page,
+              load: false,
+            },
+          }
         }
+
       }
       case 'GET_LIST_HOTEL_FAIL': {
         const { error } = action.payload;
@@ -100,7 +128,7 @@ const initialState = {
           },
         }
       }
-
+      
       case 'GET_LIST_ROOM_REQUEST': {
         return {
           ...state,
@@ -132,7 +160,7 @@ const initialState = {
           },
         }
       }
-
+      
       case 'GET_CATEGORY_LIST_REQUEST': {
         return {
           ...state,
@@ -142,7 +170,7 @@ const initialState = {
           },
         }
       }
-
+      
       case 'GET_CATEGORY_LIST_SUCCESS': {
         const { data } = action.payload;
         return {
@@ -154,7 +182,7 @@ const initialState = {
           },
         }
       }
-
+      
       case 'GET_CATEGORY_LIST_FAIL': {
         const { error } = action.payload;
         return {
@@ -169,31 +197,30 @@ const initialState = {
       case 'GET_RATE_LIST_REQUEST': {
         return {
           ...state,
-         rateList: {
+          rateList: {
             ...state.rateList,
             load: true,
           },
         }
       }
-
+      
       case 'GET_RATE_LIST_SUCCESS': {
         const { data } = action.payload;
-        console.log("🚀 ~ file: product.reducer.js ~ line 181 ~ productHotelReducer ~ data", data)
         return {
           ...state,
-         rateList: {
+          rateList: {
             ...state.rateList,
             data: data,
             load: false,
           },
         }
       }
-
+      
       case 'GET_RATE_LIST_FAIL': {
         const { error } = action.payload;
         return {
           ...state,
-         rateList: {
+          rateList: {
             ...state.rateList,
             load: false,
             error: error,
@@ -202,7 +229,6 @@ const initialState = {
       }
       case 'BOOKING_HOTEL_ROOM_SUCCESS': {
         const { data } = action.payload;
-        console.log("🚀 ~ file: product.reducer.js ~ line 166 ~ productHotelReducer ~ data", data)
         return {
           ...state,
           listRoom: {
@@ -217,6 +243,39 @@ const initialState = {
           },
         }
       }
+      case 'GET_LIST_HOTEL_BY_RATE_REQUEST': {
+        return {
+          ...state,
+          listHotel: {
+            ...state.listHotel,
+            load: true,
+          },
+        }
+      }
+      case 'GET_LIST_HOTEL_BY_RATE_SUCCESS': {
+        const { data } = action.payload;
+        return {
+          ...state,
+          listHotel: {
+            ...state.listHotel,
+            data: data,
+            load: false,
+          },
+        }
+      }
+      case 'GET_LIST_HOTEL_BY_RATE_FAIL': {
+        const { error } = action.payload;
+        return {
+          ...state,
+          listHotel: {
+            ...state.listHotel,
+            load: false,
+            error: error,
+          },
+        }
+      }
+      
+      
       default: {
         return state;
       }
diff --git a/src/redux/sagas/product.saga.js b/src/redux/sagas/product.saga.js
index 9596fdf..fe9109a 100644
--- a/src/redux/sagas/product.saga.js
+++ b/src/redux/sagas/product.saga.js
@@ -37,20 +37,18 @@ function* getProductHotelListSaga(action) {
 function* getListHotelSaga(action) {
 
   try {
-    const { id,more,page,limit,rateId } = action.payload;
+    const { id,more,page,limit,rate } = action.payload;
     const result = yield axios({
       method: 'GET',
-      url: `http://localhost:3002/locations/${id}`,
+      url: `http://localhost:3002/locations/${id}/hotels`,
       params: {
-        _embed: "hotels",
-        page: 1,
-        limit: 4,
-        ...rateId && {rateId}
-
-        
-        
+        _page: page,
+        _limit: limit,
+        ...rate && { rate },
       }
     });
+    console.log("🚀 ~ file: product.saga.js ~ line 59 ~ function*getListHotelSaga ~ action.payload", action.payload)
+    console.log("🚀 ~ file: product.saga.js ~ line 60 ~ function*getListHotelSaga ~ result", result)
     
     yield put({
       type: "GET_LIST_HOTEL_SUCCESS",
@@ -66,10 +64,10 @@ function* getListHotelSaga(action) {
 }
 
 function* getListRoomSaga(action) {
-
+  
   // Chỗ này e phải lấy data của hotel chứ
   // Do e đặt tên bảng lojn xộn nên gây nhầm lẫn
-
+  
   try {
     const { id } = action.payload;
     const result = yield axios({
@@ -77,7 +75,7 @@ function* getListRoomSaga(action) {
       url: `http://localhost:3002/hotels/${id}?_embed=rooms&_embed=bookingRooms&_expand=location`,
       
     });
-   
+    
     yield put({
       type: "GET_LIST_ROOM_SUCCESS",
       payload: {
@@ -117,24 +115,24 @@ function* getRateListSaga(action) {
       method: 'GET',
       url: 'http://localhost:3002/rates',
       // params: {
-      //   _embed: "hotels", 
-      // }
-    });
-    yield put({
-      type: "GET_RATE_LIST_SUCCESS",
-      payload: {
-        data: result.data,
-      },
-    });
-  } catch (e) {
-    yield put({
-      type: "GET_RATE_LIST_FAIL",
-      payload: {
-        error: e.error
-      },
-    });
+        //   _embed: "hotels", 
+        // }
+      });
+      yield put({
+        type: "GET_RATE_LIST_SUCCESS",
+        payload: {
+          data: result.data,
+        },
+      });
+    } catch (e) {
+      yield put({
+        type: "GET_RATE_LIST_FAIL",
+        payload: {
+          error: e.error
+        },
+      });
+    }
   }
-}
 
 export default function* productHotelSaga() {
   yield takeEvery('GET_PRODUCT_HOTEL_LIST_REQUEST', getProductHotelListSaga);
